src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -11,25 +11,48 @@
   (not= (get-in req1 [:params "query"])
         (get-in req2 [:params "query"])))
 
-(defn add-criteria [crit query]
+(defn query-criteria
+  "Extract the 'criteria' (select) part of the given query"
+  [query]
   (cm/match [query]
-            [["extract" columns (expr :guard nil?)]]
-            ["extract" columns crit]
+    [["extract" _ expr]] expr
+    :else (or query nil)))
+
+(defn is-active-node-criteria? [criteria]
+  (cm/match [criteria]
+    [["=" ["node" "active"] _]] criteria
+    :else false))
```

Comment:
we also support queries like ["null?", "expired", true/false] . Should we be picking those up as well here?